### PR TITLE
Impl docservice

### PR DESCRIPTION
**Pull request by RaduCStefanescu**
Update document service interface with more appropriate methods.
Implement the API.
Use consolidated document service to manage operations on documents.

### CODE REVIEW

File: src/main/java/com/code4ro/legalconsultation/common/builders/DocumentMetadataBuilder.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package com.code4ro.legalconsultation.common.builders;
+
+import com.code4ro.legalconsultation.model.dto.DocumentView;
+import com.code4ro.legalconsultation.model.persistence.DocumentMetadata;
+import com.code4ro.legalconsultation.model.persistence.DocumentType;
+
+import java.math.BigInteger;
+import java.util.Date;
+
+public class DocumentMetadataBuilder implements GenericBuilder<DocumentMetadata> {
```
Comment (mihairinzis) [inline src/main/java/com/code4ro/legalconsultation/common/builders/DocumentMetadataBuilder.java]: we also have the `@Builder` annotation from `Lombok` :)

File: src/main/java/com/code4ro/legalconsultation/service/impl/DocumentServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        DocumentBreakdown breakdown = documentBreakdownService.build(Paths.get(filePath));
+
+        return documentConsolidatedService.saveOne(new DocumentConsolidated(metadata, breakdown));
+    }
+
+    @Override
+    public Optional<DocumentConsolidated> update(final String id, final DocumentView document, final MultipartFile file) {
+        Optional<DocumentConsolidated> consolidated = documentConsolidatedService.findOne(id);
+
+        if(consolidated.isPresent()){
```
Comment (mihairinzis) [inline src/main/java/com/code4ro/legalconsultation/service/impl/DocumentServiceImpl.java]: you can invert if here in order to reduce indentation:
```
if(!consolidated.isPresent()) {
    return Optional.empty();
}
```

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -4,7 +4,7 @@ spring:
   datasource:
     url: jdbc:mysql://localhost:3306/legalconsultation?serverTimezone=Europe/Bucharest
     username: root
-    password: root
+    password: CCTech#123
```
Comment (mihairinzis) [inline src/main/resources/application.yml]: local password

File: src/main/java/com/code4ro/legalconsultation/controller/DocumentController.java [hunk 4]
```diff
@@ -1,68 +1,89 @@
...
+        if(optDocument.isPresent())
+            return ResponseEntity.ok(optDocument.get());
+        else
+            return ResponseEntity.notFound().build();
+    }
+
+    @GetMapping("/{id}/consolidated")
+    public ResponseEntity getDocumentConsolidatedById(@PathVariable String id){
+        Optional<DocumentConsolidated> consolidatedDocumentOpt = documentService.fetchOneConsolidated(id);
+        if (consolidatedDocumentOpt.isPresent()) {
```
Comment (aggo) [inline src/main/java/com/code4ro/legalconsultation/controller/DocumentController.java]: We could throw exceptions here and let the GlobalExceptionHandler decide what kind of response is suitable.

**Pull request merged**